autogluon/utils/tabular/ml/models/lr/lr_model.py

```diff
@@ -0,0 +1,197 @@
+import logging
+import re
+
+import numpy as np
+from pandas import DataFrame
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import FeatureUnion, Pipeline
+from sklearn.preprocessing import StandardScaler, QuantileTransformer
+
+from autogluon.utils.tabular.ml.constants import BINARY, REGRESSION
+from autogluon.utils.tabular.ml.models.abstract.abstract_model import AbstractModel
+from autogluon.utils.tabular.ml.models.lr.hyperparameters.parameters import get_param_baseline, get_model_params, get_default_params, INCLUDE, IGNORE, ONLY
+from autogluon.utils.tabular.ml.models.lr.hyperparameters.searchspaces import get_default_searchspace
+from autogluon.utils.tabular.ml.models.lr.lr_preprocessing_utils import NlpDataPreprocessor, OheFeaturesGenerator, NumericDataPreprocessor
+
+logger = logging.getLogger(__name__)
+
+
+class LinearModel(AbstractModel):
+
+    def __init__(self, path: str, name: str, problem_type: str, objective_func, hyperparameters=None, features=None,
```

Comment:
Should be changed to:

```
    def __init__(self, path: str, name: str, problem_type: str, objective_func, hyperparameters=None, features=None,
                 feature_types_metadata=None, debug=0, **kwargs):
        self.model_class, self.penalty, self.handle_text = get_model_params(problem_type, hyperparameters)
        name = name + '-' + self.model_class.__name__
        name = name + '-text_' + self.handle_text
        super().__init__(path=path, name=name, problem_type=problem_type, objective_func=objective_func, hyperparameters=hyperparameters, features=features,
                         feature_types_metadata=feature_types_metadata, debug=debug)
        self.types_of_features = None
        self.pipeline = None

        self.model_params, default_params = get_default_params(self.problem_type, self.penalty)
        for param, val in default_params.items():
            self._set_default_param_value(param, val)
```

Otherwise self.name and self.path are misaligned and the model is saved in the root directory "." rather than in {dir}/models/

Reply:
good catch